### ISSUE TITLE
Error starting gRPC call. HttpRequestException: Name or service not known

### ISSUE THREAD

**Issue by rt-13** [grpc/grpc-dotnet#1366]
I have a microservice application built on .NET 5.0. I have multiple gprc and web api services. The services are deployed on Azure AKS (linux) with Ingress (nginx) controller (have 2 different ingress controllers i.e. one for all web apis and one all grpcs) and context based routing. These services communicate over https.

I am getting below error when I call any grpc service from the web api deployed to AKS. When I run the web apis locally and connect to grpc deployed to AKS, it works fine.

**Error starting gRPC call. HttpRequestException: Name or service not known (myapi.mydomain.com:443) SocketException: Name or service not known", DebugException="System.Net.Http.HttpRequestException: Name or service not known (myapi.mydomain.com:443)**

Because I have configured context based routing, I am using subdirectory delegate handler as recommended in Microsoft docs.

```
public class SubdirectoryHandler : DelegatingHandler
{
	private readonly string _subdirectory;
	public SubdirectoryHandler(HttpClientHandler innerHandler, string subdirectory) : base(innerHandler)
	{
		_subdirectory = subdirectory;
		_logger = logger;
	}

	protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
	{
		var url = $"{request.RequestUri.Scheme}://{request.RequestUri.Host}{_subdirectory}{request.RequestUri.AbsolutePath}";
		request.RequestUri = new Uri(url, UriKind.Absolute);
		return base.SendAsync(request, cancellationToken);
	}
}
```
I have also added the code to bypass server certificate validation.

```
var httpHandler = new HttpClientHandler
{
	ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator
};
var handler = new SubdirectoryHandler(httpHandler, "/grpcapp");
var httpClient = new HttpClient(handler);
var channel = GrpcChannel.ForAddress("https://myapi.mydomain.com", new GrpcChannelOptions { HttpClient = httpClient });
return channel;
```

I am not sure what I am missing. Please help.

**Comment by JamesNK** [comment]
I'm not familiar with this error. It is coming from the underlying networking layer, which is failing to find your server.

Is the server address different when deployed?

Have you looked at the search results for this error:
https://www.google.com/search?q=System.Net.Http.HttpRequestException%3A+Name+or+service+not+known

For example, this person has the same error when deploying to Docker containers: https://stackoverflow.com/questions/63712606/socketexception-name-or-service-not-known-when-url-loads-fine-in-browser

**Comment by rt-13** [comment]
I had looked into the results you have posted. But weren't really helpful. I have configured all my services in Azure Kubernetes Service with Ingress controller for SSL. So the services are accessible using a DNS url, I can't be accessing them using container name, etc. Like I said, when I connect to the gprc services from local machine, they seem to work fine. But when trying to connect from Web API hosted in same cluster as grpc service, I have been getting the above error. This makes me believe the issue is with the client configuration I have web api making the grpc call.

Anyway, I have removed the grpc service from behind the Ingress controller and have deployed as separate workloads without Ingress and SSL. This is working now, though not as per the original deployment architecture.

Thank you again for your help.
